crates/host/src/metrics.rs

```diff
@@ -20,12 +34,37 @@ pub struct HostMetrics {
     // Eventually a host will be able to support multiple lattices, so this will need to either be
     // removed or metrics will need to be scoped per-lattice.
     pub lattice_id: String,
+
+    _refresh_task_handle: Arc<RefreshWrapper>,
+}
+
+struct SystemMetrics {
+    system_total_memory_bytes: u64,
+    /// The total amount of used system memory in bytes.
+    system_used_memory_bytes: u64,
+    /// The total cpu usage.
+    system_cpu_usage: f64,
+}
+
+/// A helper struct for encapsulating the system metrics that should be wrapped in an Arc. When the
+/// final reference is removed, the drop will abort the watch task. This allows the metrics to be
+/// clonable
+#[derive(Debug)]
+struct RefreshWrapper(JoinHandle<()>);
+
+impl Drop for RefreshWrapper {
+    fn drop(&mut self) {
+        self.0.abort();
+    }
 }
 
 impl HostMetrics {
     /// Construct a new [`HostMetrics`] instance for accessing the various wasmcloud host metrics linked to the provided meter.
-    #[must_use]
-    pub fn new(meter: &Meter, host_id: String, lattice_id: String) -> Self {
+    pub fn new(
+        meter: &Meter,
+        host_id: String,
+        lattice_id: String,
+    ) -> anyhow::Result<Self, anyhow::Error> {
```

Comment:
```suggestion
    ) -> anyhow::Result<Self> {
```

NIT: probably this works?